contrib/OpenInExplorer.lua

```diff
@@ -32,82 +33,70 @@ Select the photo(s) you wish to find in explorer and press "Go to Folder".
 A file explorer window will be opened for each selected file at the file's location; the file will be highlighted.
 
 ----KNOWN ISSUES----
+Under macOS the file manager Finder opens only once and selects the file name of the last in DT selected image.
 ]]
 
 local dt = require "darktable"
 local du = require "lib/dtutils"
 local df = require "lib/dtutils.file"
 local dsys = require "lib/dtutils.system"
+local gettext = dt.gettext
 
 --Check API version
 du.check_min_api_version("5.0.0", "OpenInExplorer") 
 
-local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+gettext.bindtextdomain("OpenInExplorer",dt.configuration.config_dir.."/lua/locale/")
 
---Detect OS and modify accordingly--	
-local proper_install = false
-if dt.configuration.running_os ~= "macos" then
-  proper_install = true
-else
-  dt.print_error('OpenInExplorer plug-in only supports Windows and Linux at this time')
-  return
+local function _(msgid)
+    return gettext.dgettext("OpenInExplorer", msgid)
 end
 
+local act_os = dt.configuration.running_os
+local PS = act_os == "windows" and  "\\"  or  "/"
 
--- FUNCTIONS --
-
-local function open_in_explorer() --Open in Explorer
-  local images = dt.gui.selection()
-  local curr_image = ""
-  if #images == 0 then
-    dt.print('please select an image')
-  elseif #images <= 15 then
-    for _,image in pairs(images) do 
-      curr_image = image.path..PS..image.filename
-      local run_cmd = "explorer.exe /select, "..curr_image
-      dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
-      resp = dsys.external_command(run_cmd)
-    end
-  else
-    dt.print('please select fewer images (max 15)')
-  end
+--Detect OS and quit if not one of macOS, Windows or Linux--	
+local proper_install = true
+if act_os ~= "macos" and act_os ~= "windows" and act_os ~= "linux" then
+  proper_install = false
+  dt.print_error(_('OpenInExplorer plug-in only supports macOS, Windows and Linux at this time'))
+  return
 end
 
-local function open_in_nautilus() --Open in Nautilus
+--The commands to open the OSes' file managers
+local fmng_cmd = {}
+fmng_cmd.linux = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass 1 ]]
+fmng_cmd.macos = 'open -R '
+fmng_cmd.windows = 'explorer.exe /select, '
+
+--The working function that opens the file manager with the image file name selected.
+local function open_in_fmanager(os, fmcmd)

```

Comment:
I'm not sure that passing os as a variable is a good idea, since I think that would override the lua supplied os.xxx() functions.

Reply:
Are lua function parameter variables not local by default? But I won't take any risk, I'll name this parameter op_sys.

Now I want to commit my ultimate script. How should I proceed? I'm somewhat helpless at the sight of forks and branches and PRs. Many thanks for your help.

Reply:
You should just be able to push the changes to your branch on github and they should update the pull request

Reply:
Just a few questions:

I'd like to add a few lines to the USAGE section about the differences concerning the actual OS. On the Mac there is one Finder window for each directory the image files belong to. In these windows the last corresponding selected image is highlighted. Is it different with Explorer and Nautilus?

Is there a way to translate this USAGE section?

Why isn't the variable "resp" local that holds the dsys.external_command return value? Is there a global variabe (which I couldn't find)? I think I can dispense with it completely and change the lines in question to simple statements as I don't need the value. To add a line with dt.print_error in case the command fails, is not really necessary, for you can conclude this fact from the missing output.

Reply:
I updated my old Windows notebook and found that separate Explorer windows are opened for each selected image. My even older Linux notebook will possibly be a greater challange :-/

Reply:
Explorer and Nautilus open one window for each image and the image is highlighted. 

There is no way to translate the usage section as it currently stands, since it's the leading comments to the file before any functions are defined.  script_manager displays the leading comments when you hover over the enable/disable button, and it gets them by grabbing the second block of comments from the file.  The libraries have built in documentation that can be accessed to build the library docs, but that's not set up for translation either.  Perhaps we need to open an issue about how to implement online help so that we could do usage translation.

The resp variable should be local.  It just returns success or failure from the command.

Linux will work the same as windows, i.e. 1 window per image.  I run linux as my base os, then run windows and macos in Virtualbox VMs.  I keep just enough images in each of the VMs to do testing.

Reply:
Whoopee, my Linux notebook still obeys like an old dog, but leisurely as it is of a similar age. When I used to work on a Linux desktop, I also had a VB Windows 7 VM. But I don‘t need Windows any more, and my iMac lets me execute nearly all of the programs I was used to on Linux. As a benefit I get excellent communication with our family iPads and iPhones. So I‘ll save disk space and refrain from installing a second os.

My tests on openSUSE with KDE GUI featuring Dolphin as the default file manager showed one Dolphin window with one tab for each respective directory, highlighting all of the selected files within. In my opinion this is the optimal presentation.

I made a request on macuser, if someone knows of a similar behavior for macOS Finder.

Have a good time, don‘t let COVID-19 catch you. Volker

Reply:
OT: Apropos whoopee: A wonderful definition of photography by an American photographer whose name I unfortunately forgot: "Photography is a way to say whoopee again." I think of it each time I work on a photo in dt.